### PR TITLE
Add retries to helix failures and re-enable helix as a required workload

### PR DESCRIPTION
**Pull request by jkotalik**
Trying to see if max retries works correctly and if we can start making helix a required check. Still prototyping.

### CODE REVIEW

**Comment by jkotalik** [comment]
Interop.FunctionalTests has been failing to build for a while. Investigating.

File: src/Http/Http.Abstractions/test/PathStringTests.cs [hunk 1]
```diff
@@ -49,6 +49,22 @@ public void NotEquals_EmptyPathStringAndNonNullPathString()
             Assert.NotEqual(pathString, PathString.Empty);
         }
 
+        [Fact]
```
Comment (dougbu) [inline src/Http/Http.Abstractions/test/PathStringTests.cs]: Change here won't be checked in. Correct?

Comment (jkotalik) [reply src/Http/Http.Abstractions/test/PathStringTests.cs]: They are already removed.

**Comment by anurse** [comment]
@aspnet/build, @HaoK FYI this is part of exploring things we can do to get Helix up and running and reliable for our scenarios.

**Review by dougbu — APPROVED** [review]
If it passes :shipit:

**Review by HaoK — APPROVED** [review]
Seems strange to have a separate flaky run and also retry failing tests 4 times, but otherwise looks good, there seems to only be a handful of tests that were frequent offenders, so hopefully retries will be enough

**Comment by anurse** [comment]
> Seems strange to have a separate flaky run and also retry failing tests 4 times, but otherwise looks good

I entirely agree. If this is a feasible approach, then I would definitely consider doing it over `FlakyAttribute`. Particularly since my understanding is that even if it succeeds on retry, Helix will still alert us of the failure (somehow...?)

**Comment by jkotalik** [comment]
> Helix will still alert us of the failure (somehow...?)

It would look like this: https://github.com/dotnet/corefx/pull/36575#issuecomment-480146504

It requires people to look at helix though even if there is a green checkmark. I think there are a few things we can do to improve the experience with helix (to get in line with what corefx does).

**Comment by jkotalik** [comment]
Going to let @aspnet/build decide how to approach re-enabling helix for now. I already brain dumped to @natemcmaster some ideas to make builds and tests fast.

**Review by natemcmaster — APPROVED** [review]

**Comment by jkotalik** [comment]
Going to merge this later today. I think we need https://github.com/aspnet/AspNetCore/pull/9505 before merging.

**Comment by jkotalik** [comment]
@aspnet/build there seems to be an agent issue with a Debian 8 agent on this test run: 
```

Starting test execution, please wait...
[xUnit.net 00:00:04.91]     Microsoft.AspNetCore.SignalR.Client.FunctionalTests.HubProtocolVersionTests.ClientUsingOldCallWithNewProtocol(transportType: LongPolling) [FAIL]
Failed   Microsoft.AspNetCore.SignalR.Client.FunctionalTests.HubProtocolVersionTests.ClientUsingOldCallWithNewProtocol(transportType: LongPolling)
Error Message:
 System.Exception : 1 error(s) logged.
Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingProvider - 48 - An error occurred while reading the key ring.
===================
System.Xml.XmlException: Root element is missing.
   at System.Xml.XmlTextReaderImpl.Throw(Exception e)
   at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
   at System.Xml.XmlReader.MoveToContent()
   at System.Xml.Linq.XElement.Load(XmlReader reader, LoadOptions options)
   at System.Xml.Linq.XElement.Load(Stream stream, LoadOptions options)
   at Microsoft.AspNetCore.DataProtection.Repositories.FileSystemXmlRepository.ReadElementFromFile(String fullPath) in /_/src/DataProtection/DataProtection/src/Repositories/FileSystemXmlRepository.cs:line 102
   at Microsoft.AspNetCore.DataProtection.Repositories.FileSystemXmlRepository.GetAllElementsCore()+MoveNext() in /_/src/DataProtection/DataProtection/src/Repositories/FileSystemXmlRepository.cs:line 83
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Microsoft.AspNetCore.DataProtection.Repositories.FileSystemXmlRepository.GetAllElements() in /_/src/DataProtection/DataProtection/src/Repositories/FileSystemXmlRepository.cs:line 68
   at Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager.GetAllKeys() in /_/src/DataProtection/DataProtection/src/KeyManagement/XmlKeyManager.cs:line 152
   at Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingProvider.CreateCacheableKeyRingCore(DateTimeOffset now, IKey keyJustAdded) in /_/src/DataProtection/DataProtection/src/KeyManagement/KeyRingProvider.cs:line 57
   at Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingProvider.GetCurrentKeyRingCore(DateTime utcNow) in /_/src/DataProtection/DataProtection/src/KeyManagement/KeyRingProvider.cs:line 177
   at Microsoft.AspNetCore.DataProtection.Internal.DataProtectionStartupFilter.Configure(Action`1 next) in /_/src/DataProtection/DataProtection/src/Internal/DataProtectionStartupFilter.cs:line 34
===================
Stack Trace:
   at Microsoft.AspNetCore.SignalR.Tests.VerifyNoErrorsScope.Dispose() in F:\workspace\_work\1\s\src\SignalR\common\testassets\Tests.Utils\VerifyNoErrorsScope.cs:line 61
   at Microsoft.AspNetCore.SignalR.Tests.FunctionalTestBase.MultiDisposable.Dispose() in F:\workspace\_work\1\s\src\SignalR\common\testassets\Tests.Utils\FunctionalTestBase.cs:line 55
   at Microsoft.AspNetCore.SignalR.Client.FunctionalTests.HubProtocolVersionTests.ClientUsingOldCallWithNewProtocol(HttpTransportType transportType) in /_/src/SignalR/clients/csharp/Client/test/FunctionalTests/HubProtocolVersionTests.cs:line 95
--- End of stack trace from previous location where exception was thrown ---
Standard Output Messages:
 | [0.012s] TestLifetime Information: Starting test ClientUsingOldCallWithNewProtocol-LongPolling at 2019-04-18T22:00:16
 | [0.123s] Microsoft.AspNetCore.SignalR.Tests.InProcessTestServer Information: Starting test server...
 | [1.823s] Microsoft.AspNetCore.Hosting.Diagnostics Debug: Hosting starting
 | [1.880s] Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager Information: User profile is available. Using '/home/helixbot/.aspnet/DataProtection-Keys' as key repository; keys will not be encrypted at rest.
 | [1.896s] Microsoft.AspNetCore.DataProtection.Repositories.FileSystemXmlRepository Debug: Reading data from file '/home/helixbot/.aspnet/DataProtection-Keys/key-82db535e-42fd-41a9-8ace-43d45780953e.xml'.
 | [1.909s] Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingProvider Error: An error occurred while reading the key ring.
 | System.Xml.XmlException: Root element is missing.
 |    at System.Xml.XmlTextReaderImpl.Throw(Exception e)
 |    at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
 |    at System.Xml.XmlReader.MoveToContent()
 |    at System.Xml.Linq.XElement.Load(XmlReader reader, LoadOptions options)
 |    at System.Xml.Linq.XElement.Load(Stream stream, LoadOptions options)
 |    at Microsoft.AspNetCore.DataProtection.Repositories.FileSystemXmlRepository.ReadElementFromFile(String fullPath) in /_/src/DataProtection/DataProtection/src/Repositories/FileSystemXmlRepository.cs:line 102
 |    at Microsoft.AspNetCore.DataProtection.Repositories.FileSystemXmlRepository.GetAllElementsCore()+MoveNext() in /_/src/DataProtection/DataProtection/src/Repositories/FileSystemXmlRepository.cs:line 83
 |    at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
 |    at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
 |    at Microsoft.AspNetCore.DataProtection.Repositories.FileSystemXmlRepository.GetAllElements() in /_/src/DataProtection/DataProtection/src/Repositories/FileSystemXmlRepository.cs:line 68
 |    at Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager.GetAllKeys() in /_/src/DataProtection/DataProtection/src/KeyManagement/XmlKeyManager.cs:line 152
 |    at Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingProvider.CreateCacheableKeyRingCore(DateTimeOffset now, IKey keyJustAdded) in /_/src/DataProtection/DataProtection/src/KeyManagement/KeyRingProvider.cs:line 57
 |    at Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingProvider.GetCurrentKeyRingCore(DateTime utcNow) in /_/src/DataProtection/DataProtection/src/KeyManagement/KeyRingProvider.cs:line 177
 | [1.922s] Microsoft.AspNetCore.DataProtection.Internal.DataProtectionStartupFilter Information: Key ring failed to load during application startup.
 | System.Xml.XmlException: Root element is missing.
 |    at System.Xml.XmlTextReaderImpl.Throw(Exception e)
 |    at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
 |    at System.Xml.XmlReader.MoveToContent()
 |    at System.Xml.Linq.XElement.Load(XmlReader reader, LoadOptions options)
 |    at System.Xml.Linq.XElement.Load(Stream stream, LoadOptions options)
 |    at Microsoft.AspNetCore.DataProtection.Repositories.FileSystemXmlRepository.ReadElementFromFile(String fullPath) in /_/src/DataProtection/DataProtection/src/Repositories/FileSystemXmlRepository.cs:line 102
 |    at Microsoft.AspNetCore.DataProtection.Repositories.FileSystemXmlRepository.GetAllElementsCore()+MoveNext() in /_/src/DataProtection/DataProtection/src/Repositories/FileSystemXmlRepository.cs:line 83
 |    at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
 |    at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
 |    at Microsoft.AspNetCore.DataProtection.Repositories.FileSystemXmlRepository.GetAllElements() in /_/src/DataProtection/DataProtection/src/Repositories/FileSystemXmlRepository.cs:line 68
 |    at Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager.GetAllKeys() in /_/src/DataProtection/DataProtection/src/KeyManagement/XmlKeyManager.cs:line 152
 |    at Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingProvider.CreateCacheableKeyRingCore(DateTimeOffset now, IKey keyJustAdded) in /_/src/DataProtection/DataProtection/src/KeyManagement/KeyRingProvider.cs:line 57
 |    at Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingProvider.GetCurrentKeyRingCore(DateTime utcNow) in /_/src/DataProtection/DataProtection/src/KeyManagement/KeyRingProvider.cs:line 177
 |    at Microsoft.AspNetCore.DataProtection.Internal.DataProtectionStartupFilter.Configure(Action`1 next) in /_/src/DataProtection/DataProtection/src/Internal/DataProtectionStartupFilter.cs:line 34
 | [1.952s] Microsoft.AspNetCore.SignalR.Internal.DefaultHubProtocolResolver Debug: Registered SignalR Protocol: json, implemented by Microsoft.AspNetCore.SignalR.Client.FunctionalTests.VersionedJsonHubProtocol.
 | [2.088s] Microsoft.AspNetCore.SignalR.Tests.InProcessTestServer Information: Test Server started
 | [2.101s] Microsoft.AspNetCore.Hosting.Diagnostics Debug: Hosting started
 | [2.101s] Microsoft.AspNetCore.Hosting.Diagnostics Debug: Loaded hosting startup assembly Microsoft.AspNetCore.SignalR.Client.FunctionalTests
 | [2.127s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Starting HubConnection.
 | [2.165s] Microsoft.AspNetCore.Http.Connections.Client.HttpConnection Debug: Starting HttpConnection.
 | [2.195s] Microsoft.AspNetCore.Http.Connections.Client.HttpConnection Debug: Establishing connection with server at 'http://127.0.0.1:53077/version'.
 | [2.312s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLM494VID4ER" started.
 | [2.525s] Microsoft.AspNetCore.Hosting.Diagnostics Information: Request starting HTTP/1.1 POST http://127.0.0.1:53077/version/negotiate  0
 | [2.592s] Microsoft.AspNetCore.Routing.RouteBase Debug: Request successfully matched the route with name '(null)' and template '/version/negotiate'
 | [2.627s] Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionManager Debug: New connection Z51kCizUKpjlqsuB7yES5g created.
 | [2.655s] Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionDispatcher Debug: Z51kCizUKpjlqsuB7yES5g - Sending negotiation response.
 | [2.743s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLM494VID4ER" completed keep alive response.
 | [2.765s] Microsoft.AspNetCore.Hosting.Diagnostics Information: Request finished in 239.85350000000003ms 200 application/json
 | [2.811s] Microsoft.AspNetCore.Http.Connections.Client.HttpConnection Debug: Established connection 'Z51kCizUKpjlqsuB7yES5g' with the server.
 | [2.819s] Microsoft.AspNetCore.Http.Connections.Client.HttpConnection Debug: Skipping transport WebSockets because it was disabled by the client.
 | [2.819s] Microsoft.AspNetCore.Http.Connections.Client.HttpConnection Debug: Skipping transport ServerSentEvents because it was disabled by the client.
 | [2.820s] Microsoft.AspNetCore.Http.Connections.Client.HttpConnection Debug: Starting transport 'LongPolling' with Url: http://127.0.0.1:53077/version?id=Z51kCizUKpjlqsuB7yES5g.
 | [2.830s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Information: Starting transport. Transfer mode: Text.
 | [2.840s] Microsoft.AspNetCore.Hosting.Diagnostics Information: Request starting HTTP/1.1 GET http://127.0.0.1:53077/version?id=Z51kCizUKpjlqsuB7yES5g  
 | [2.840s] Microsoft.AspNetCore.Routing.RouteBase Debug: Request successfully matched the route with name '(null)' and template '/version'
 | [2.898s] Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionDispatcher Debug: Z51kCizUKpjlqsuB7yES5g - Establishing new connection.
 | [2.905s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLM494VID4ER" completed keep alive response.
 | [2.907s] Microsoft.AspNetCore.Hosting.Diagnostics Information: Request finished in 66.5209ms 200 application/octet-stream
 | [2.911s] Microsoft.AspNetCore.SignalR.HubConnectionHandler Debug: Z51kCizUKpjlqsuB7yES5g - OnConnectedAsync started.
 | [2.939s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Starting receive loop.
 | [2.947s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Starting the send loop.
 | [2.954s] Microsoft.AspNetCore.Http.Connections.Client.HttpConnection Debug: Transport 'LongPolling' started.
 | [2.961s] Microsoft.AspNetCore.Http.Connections.Client.HttpConnection Information: HttpConnection Started.
 | [2.961s] Microsoft.AspNetCore.Hosting.Diagnostics Information: Request starting HTTP/1.1 GET http://127.0.0.1:53077/version?id=Z51kCizUKpjlqsuB7yES5g  
 | [2.962s] Microsoft.AspNetCore.Routing.RouteBase Debug: Request successfully matched the route with name '(null)' and template '/version'
 | [2.963s] Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionDispatcher Debug: Z51kCizUKpjlqsuB7yES5g - Resuming existing connection.
 | [2.983s] Microsoft.AspNetCore.SignalR.Client.HubConnection Information: Using HubProtocol 'json v1000'.
 | [2.990s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Sending Hub Handshake.
 | [2.998s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Sending 35 bytes to the server using url: http://127.0.0.1:53077/version?id=Z51kCizUKpjlqsuB7yES5g.
 | [3.002s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLM494VID4ET" started.
 | [3.006s] Microsoft.AspNetCore.Hosting.Diagnostics Information: Request starting HTTP/1.1 POST http://127.0.0.1:53077/version?id=Z51kCizUKpjlqsuB7yES5g  35
 | [3.006s] Microsoft.AspNetCore.Routing.RouteBase Debug: Request successfully matched the route with name '(null)' and template '/version'
 | [3.024s] Microsoft.AspNetCore.Server.Kestrel Debug: Z51kCizUKpjlqsuB7yES5g - Connection id "0HLM494VID4ET", Request id "0HLM494VID4ET:00000001": started reading request body.
 | [3.026s] Microsoft.AspNetCore.Server.Kestrel Debug: Z51kCizUKpjlqsuB7yES5g - Connection id "0HLM494VID4ET", Request id "0HLM494VID4ET:00000001": done reading request body.
 | [3.035s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLM494VID4ET" completed keep alive response.
 | [3.035s] Microsoft.AspNetCore.Hosting.Diagnostics Information: Request finished in 29.289ms 200 text/plain
 | [3.046s] Microsoft.AspNetCore.SignalR.Internal.DefaultHubProtocolResolver Debug: Z51kCizUKpjlqsuB7yES5g - Found protocol implementation for requested protocol: json.
 | [3.063s] Microsoft.AspNetCore.SignalR.HubConnectionContext Information: Z51kCizUKpjlqsuB7yES5g - Completed connection handshake. Using HubProtocol 'json'.
 | [3.114s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLM494VID4ER" completed keep alive response.
 | [3.114s] Microsoft.AspNetCore.Hosting.Diagnostics Information: Request finished in 152.82590000000002ms 200 application/octet-stream
 | [3.124s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Message(s) sent successfully.
 | [3.132s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Received messages from the server.
 | [3.168s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Handshake with server complete.
 | [3.168s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Receive loop starting.
 | [3.168s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Sending PingMessage message.
 | [3.168s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Sending PingMessage message completed.
 | [3.169s] Microsoft.AspNetCore.SignalR.Client.HubConnection Information: HubConnection started.
 | [3.169s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Sending 11 bytes to the server using url: http://127.0.0.1:53077/version?id=Z51kCizUKpjlqsuB7yES5g.
 | [3.170s] Microsoft.AspNetCore.Hosting.Diagnostics Information: Request starting HTTP/1.1 GET http://127.0.0.1:53077/version?id=Z51kCizUKpjlqsuB7yES5g  
 | [3.170s] Microsoft.AspNetCore.Routing.RouteBase Debug: Request successfully matched the route with name '(null)' and template '/version'
 | [3.170s] Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionDispatcher Debug: Z51kCizUKpjlqsuB7yES5g - Resuming existing connection.
 | [3.180s] Microsoft.AspNetCore.Hosting.Diagnostics Information: Request starting HTTP/1.1 POST http://127.0.0.1:53077/version?id=Z51kCizUKpjlqsuB7yES5g  11
 | [3.180s] Microsoft.AspNetCore.Routing.RouteBase Debug: Request successfully matched the route with name '(null)' and template '/version'
 | [3.180s] Microsoft.AspNetCore.Server.Kestrel Debug: Z51kCizUKpjlqsuB7yES5g - Connection id "0HLM494VID4ET", Request id "0HLM494VID4ET:00000002": started reading request body.
 | [3.180s] Microsoft.AspNetCore.Server.Kestrel Debug: Z51kCizUKpjlqsuB7yES5g - Connection id "0HLM494VID4ET", Request id "0HLM494VID4ET:00000002": done reading request body.
 | [3.180s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLM494VID4ET" completed keep alive response.
 | [3.180s] Microsoft.AspNetCore.Hosting.Diagnostics Information: Request finished in 0.405ms 200 text/plain
 | [3.188s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Message(s) sent successfully.
 | [3.206s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Registering Invocation ID '1' for tracking.
 | [3.228s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Sending InvocationMessage message '1'.
 | [3.228s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Sending InvocationMessage message '1' completed.
 | [3.228s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Sending 75 bytes to the server using url: http://127.0.0.1:53077/version?id=Z51kCizUKpjlqsuB7yES5g.
 | [3.229s] Microsoft.AspNetCore.Hosting.Diagnostics Information: Request starting HTTP/1.1 POST http://127.0.0.1:53077/version?id=Z51kCizUKpjlqsuB7yES5g  75
 | [3.229s] Microsoft.AspNetCore.Routing.RouteBase Debug: Request successfully matched the route with name '(null)' and template '/version'
 | [3.229s] Microsoft.AspNetCore.Server.Kestrel Debug: Z51kCizUKpjlqsuB7yES5g - Connection id "0HLM494VID4ET", Request id "0HLM494VID4ET:00000003": started reading request body.
 | [3.229s] Microsoft.AspNetCore.Server.Kestrel Debug: Z51kCizUKpjlqsuB7yES5g - Connection id "0HLM494VID4ET", Request id "0HLM494VID4ET:00000003": done reading request body.
 | [3.229s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLM494VID4ET" completed keep alive response.
 | [3.230s] Microsoft.AspNetCore.Hosting.Diagnostics Information: Request finished in 0.6422ms 200 text/plain
 | [3.231s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Message(s) sent successfully.
 | [3.254s] Microsoft.AspNetCore.SignalR.Internal.DefaultHubDispatcher Debug: Z51kCizUKpjlqsuB7yES5g - Received hub invocation: InvocationMessage { InvocationId: "1", Target: "Echo", Arguments: [ Hello World! ], StreamIds: [  ] }.
 | [3.300s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLM494VID4ER" completed keep alive response.
 | [3.300s] Microsoft.AspNetCore.Hosting.Diagnostics Information: Request finished in 129.6132ms 200 application/octet-stream
 | [3.300s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Received messages from the server.
 | [3.304s] Microsoft.AspNetCore.Hosting.Diagnostics Information: Request starting HTTP/1.1 GET http://127.0.0.1:53077/version?id=Z51kCizUKpjlqsuB7yES5g  
 | [3.304s] Microsoft.AspNetCore.Routing.RouteBase Debug: Request successfully matched the route with name '(null)' and template '/version'
 | [3.304s] Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionDispatcher Debug: Z51kCizUKpjlqsuB7yES5g - Resuming existing connection.
 | [3.306s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Processing 54 byte message from server.
 | [3.338s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Stopping HubConnection.
 | [3.338s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Terminating receive loop.
 | [3.339s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Waiting for the receive loop to terminate.
 | [3.358s] Microsoft.AspNetCore.Http.Connections.Client.HttpConnection Debug: Disposing HttpConnection.
 | [3.360s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Information: Transport is stopping.
 | [3.364s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Send loop canceled.
 | [3.366s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Send loop stopped.
 | [3.369s] Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets Debug: Connection id "0HLM494VID4ER" received FIN.
 | [3.386s] Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets Debug: Connection id "0HLM494VID4ER" sending FIN because: "The client closed the connection."
 | [3.390s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Receive loop stopped.
 | [3.391s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLM494VID4ER" disconnecting.
 | [3.396s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Sending DELETE request to 'http://127.0.0.1:53077/version?id=Z51kCizUKpjlqsuB7yES5g'.
 | [3.397s] Microsoft.AspNetCore.Hosting.Diagnostics Information: Request starting HTTP/1.1 DELETE http://127.0.0.1:53077/version?id=Z51kCizUKpjlqsuB7yES5g  
 | [3.398s] Microsoft.AspNetCore.Routing.RouteBase Debug: Request successfully matched the route with name '(null)' and template '/version'
 | [3.405s] Microsoft.AspNetCore.Http.Connections.Internal.Transports.LongPollingTransport Debug: Z51kCizUKpjlqsuB7yES5g - Client disconnected from Long Polling endpoint for connection.
 | [3.405s] Microsoft.AspNetCore.Hosting.Diagnostics Information: Request finished in 101.4118ms 0 
 | [3.414s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLM494VID4ER" stopped.
 | [3.433s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLM494VID4ET" completed keep alive response.
 | [3.434s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: DELETE request to 'http://127.0.0.1:53077/version?id=Z51kCizUKpjlqsuB7yES5g' accepted.
 | [3.435s] Microsoft.AspNetCore.SignalR.HubConnectionHandler Debug: Z51kCizUKpjlqsuB7yES5g - OnConnectedAsync ending.
 | [3.435s] Microsoft.AspNetCore.Http.Connections.Client.Internal.LongPollingTransport Debug: Transport stopped.
 | [3.437s] Microsoft.AspNetCore.Http.Connections.Client.HttpConnection Information: HttpConnection Disposed.
 | [3.433s] Microsoft.AspNetCore.Hosting.Diagnostics Information: Request finished in 35.3802ms 202 text/plain
 | [3.442s] Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets Debug: Connection id "0HLM494VID4ET" received FIN.
 | [3.443s] Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets Debug: Connection id "0HLM494VID4ET" sending FIN because: "The client closed the connection."
 | [3.443s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLM494VID4ET" disconnecting.
 | [3.443s] Microsoft.AspNetCore.Server.Kestrel Debug: Connection id "0HLM494VID4ET" stopped.
 | [3.446s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: Canceling all outstanding invocations.
 | [3.448s] Microsoft.AspNetCore.SignalR.Client.HubConnection Debug: HubConnection stopped.
 | [3.451s] Microsoft.AspNetCore.SignalR.Tests.InProcessTestServer Information: Shutting down test server
 | [3.451s] Microsoft.AspNetCore.Http.Connections.Internal.HttpConnectionManager Debug: Z51kCizUKpjlqsuB7yES5g - Removing connection Z51kCizUKpjlqsuB7yES5g from the list of connections.
 | [3.458s] Microsoft.AspNetCore.Hosting.Diagnostics Debug: Hosting shutdown
 | [3.475s] Microsoft.AspNetCore.SignalR.Tests.InProcessTestServer Information: Test server shut down
 | [3.496s] TestLifetime Information: Finished test ClientUsingOldCallWithNewProtocol-LongPolling in 3.4883583999999996s
```

**Comment by natemcmaster** [comment]
Rebase on master. This one can go in now :)

**Pull request merged**